### ISSUE TITLE
Automate uploading metadata and description to stores

### ISSUE THREAD

**Issue by hellais** [TheTorProject/ooniprobe-android#125]
As pointed out in: https://github.com/TheTorProject/ooniprobe-android/pull/124#issuecomment-337154133, following suggestion from @eighthave (https://github.com/TheTorProject/ooniprobe-android/issues/87#issuecomment-337036169), we may want to use something like fastlane to automatically update app descriptions in the various languages.

This would also make updating f-droid descriptions easier (see: https://github.com/TheTorProject/ooniprobe-android/issues/87)

**Comment by lorenzoPrimi** [comment]
I've looked into fastlane and apparently it creates various directories with screenshots, text files with description, title and other images.
We would have to adapt the gen_descriptions.py script to reflect this structure, but that's not difficult.
Now counting that the screenshots can be 5 * number of languages, and so will be descriptions, would be better to have a repo only with screenshots and description (for both iOS and Android) instead of having them all inside the project repo? @hellais

**Comment by eighthave** [comment]
It would be better for Fastlane and for F-Droid to have all the Android
screenshots and materials in the same git repo as the app itself.  Both
Fastlane and F-Droid are designed around that idea.  Same with Triple-T
Gradle Play Publisher.

If you worried about having duplicate texts in the iOS and Android
repos, the translation tools handle duplicate texts very well.  They
generally will automatically fill in strings where the key is the exact
match of a previously known source string (e.g. if someone translates
Android first, then iOS).

**Comment by lorenzoPrimi** [comment]
My worry was about having all the graphics into the code repo.
I believe graphics (and descriptions) should be into another repo and not together with the code.

**Comment by eighthave** [comment]
There are already many many PNGs in your Android app's repo.  The
standard practice is to include them all in the same repo because this
makes much easier to automate.  If your descriptions and screenshots are
in a different repo, you'll have to do custom configs, and then figure
out how to make that work with F-Droid.

**Comment by hellais** [comment]
I agree with @eighthave that's is probably ideal not to have a separate repo for just assets. If we can keep it all in just one repo let's do it, otherwise let's see if we can have something as a master source of truth (maybe contentful CMS so @agrabeli can easily edit the copy) and ensure everything is always automatically generated.

**Comment by eighthave** [comment]
If you follow the standard Fastlane layout for screenshots and commit them to the app's git repo, F-Droid will automatically pick them up.

FYI, F-Droid will strip EXIF metadata from images and might "crunch" the PNGs.  I think Google Play does something similar.  If you want to have complete control over the images, you should  pre-strip and pre-crunch your images, e.g.:

```console
$ find metadata/ -name '*.jp*g' -o -name '*.png' | xargs exiftool -all=
```

Here's a script I wrote to do this, you can use it under any free software license you need:
https://gitlab.com/fdroid/fdroiddata/blob/master/tools/png-pre-compress
